.github/workflows/pr_checks.yaml

```diff
@@ -101,6 +137,7 @@ jobs:
       - envtest-tests
       - integration-tests
       - conformance-tests
-    if: ${{ always() && needs.should-run-with-secrets.outputs.result == 'true' }}
+      - up-to-date
+    if: ${{ always() && needs.should-run-with-secrets.outputs.result == 'true' && needs.up-to-date.outputs.status == 'false' }}
```

Comment:
Do we really need the `always()` here?

Reply:
This ensures that this runs even if the dependent jobs (from the `needs` section) get cancelled.

Reply:
NIT: In every `if` you can drop `${{    ...    }}` parenthesis